### ISSUE TITLE
`vue/no-ref-as-operand` to support `ref` from composable

### ISSUE THREAD

**Issue by kong-troop** [vuejs/eslint-plugin-vue#2519]
**Please describe what the rule should do:**

Current `vue/no-ref-as-operand` rule only support the local `ref`. If a `ref` is being returned from a composable and being use with operand (eg: returnedRef++) , it didnt throw any warning as expected.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```ts
const ok = ref(true)
const msg = ok ? 'yes' : 'no' // this throws warning as expected

// ---- //

const ok = useOk() // returned as Ref<boolean>
const msg = ok ? 'yes' : 'no' // No warning was shown
```

**Additional context**

**Comment by FloEdelmann** [comment]
ESLint and thus eslint-plugin-vue only has access to the current file while linting, so checking what might be returned from some different file is difficult.

Maybe if typescript-eslint's parser is used, we have access to the types and can report those cases then? Feel free to try that and submit a PR.

**Comment by Flareonn** [comment]
I have a similar problem. Based on the comment above, for the test I made a composable in one file

My example of playback:
```typescript
import { defineComponent, ref } from 'vue'

const useMyRef = () => ref(true)

export const TestComponent = defineComponent({
	name: 'TestComponent',
	setup() {
		const test2 = ref(true)
		const ctx = useMyRef()

		// Not valid - true
		if (test2) {
			console.log('')
		}

		// Valid - wtf?
		if (ctx) {
			console.log('')
		}

		return () => null
	},
})
```

**Comment by johnson86tw** [comment]
Same error here. Happy to offer $10+ in crypto to support whoever fixes it

**Comment by ValentinYoushkevich** [comment]
@FloEdelmann I have a working implementation of the type-based approach you suggested here and in #2954. It reports a value when its type carries Vue's `RefSymbol` marker, so `Ref`, `ShallowRef`, `ComputedRef` and friends are caught while `MaybeRef` isn't, and it only kicks in when type information is available.

Then I noticed your WIP `no-ref-as-operand` branch from March. Are you planning to finish it, or would a PR from me be useful? Happy to go either way.

**Comment by FloEdelmann** [comment]
Oh, I forgot about that branch! Feel free to go ahead with your implementation!
